### PR TITLE
Elasticsearch 2.1

### PR DESCRIPTION
**Pull request by ersushantsood**
Hi @jmaitrehenry  I have modified the changes tested and verified with elasticsearch 2.1 , can you please verify them and then we can merge these changes to master .

### CODE REVIEW

**Comment by jmaitrehenry** [comment]
Hi @ersushantsood,
We have 2 failing tests, once those will be fixed, I will merge without problem.
Thanks for your works!

**Comment by ersushantsood** [comment]
Hi jmaitrehenry , I have a suggestion can we keep different branches for different Elasticsearch versions as there is no backward compatibility of the plugin with older versions

**Comment by jmaitrehenry** [comment]
I think it’s a good idea to keep a branch for ES1.7 but we can keep only one branch for 2.x?
And the latest elasticsearch version should be available in master I think.
What do you think?

**Comment by ersushantsood** [comment]
Yap that would be ideal .We need to start working toward looking at supporting 2.2 and 2.3 versions also.

**Comment by jmaitrehenry** [comment]
@jehuty0shift & @ersushantsood I just rebase with master. If the next build is green, I will merge into master

@jehuty0shift do you want to check the rebase before the final merge?

File: src/main/java/com/petalmd/armor/filter/AbstractActionFilter.java [hunk 1]
```diff
@@ -50,41 +50,61 @@
...
 
     @Override
-    public final void apply(final String action, final ActionRequest request, final ActionListener listener, final ActionFilterChain chain) {
+    public final void apply(Task task, final String action, final ActionRequest request, final ActionListener listener, final ActionFilterChain chain) {
         log.debug("REQUEST on node {}: {} ({}) from {}", clusterService.localNode().getName(), action, request.getClass(),
                 request.remoteAddress() == null ? "INTRANODE" : request.remoteAddress().toString());
         log.debug("Context {}", request.getContext());
         log.debug("Headers {}", request.getHeaders());
 
-        if (settings.getAsBoolean(ConfigConstants.ARMOR_ALLOW_CLUSTER_MONITOR, true) && action.startsWith("cluster:monitor/")) {
```
Comment (jmaitrehenry) [inline src/main/java/com/petalmd/armor/filter/AbstractActionFilter.java:86]: This change break testClusterMonitorDisable

File: src/main/java/com/petalmd/armor/filter/ArmorActionFilter.java [hunk 2]
```diff
@@ -125,11 +126,12 @@ private void copyContextToHeader(final ActionRequest request) {
         }
     }
 
-    private void apply0(final String action, final ActionRequest request, final ActionListener listener, final ActionFilterChain chain)
+    private void apply0(Task task, final String action, final ActionRequest request, final ActionListener listener, final ActionFilterChain chain)
             throws Exception {
-
-        if (settings.getAsBoolean(ConfigConstants.ARMOR_ALLOW_CLUSTER_MONITOR, true) && action.startsWith("cluster:monitor/")) {
```
Comment (jmaitrehenry) [inline src/main/java/com/petalmd/armor/filter/ArmorActionFilter.java:131]: And this too

**Comment by jehuty0shift** [comment]
Yep since i introduced these changes. I can fix the merge if you want. The idea behind this option was to make it possible to allow only the monitor actions that Kibana needs. Since Kibana needs more than monitors action now, i could maybe rewrite it into allow_kibana_actions ? (that will allows all monitor and kibana actions needed by Kibana).

**Comment by jmaitrehenry** [comment]
Sound good for me.

**Pull request merged**